### PR TITLE
Feature/code mirror theme exposed

### PR DESCRIPTION
**Pull request by joelgriffith**
First pass on exposing a `codeMirrorTheme` prop, plumbing it through, and docs on how to load/use css for the theme.

### CODE REVIEW

**Comment by joelgriffith** [comment]
See discussion in https://github.com/graphql/graphiql/issues/129

File: test/server.js [hunk 1]
```diff
@@ -9,7 +9,6 @@
  /* eslint-disable no-console */
 import express from 'express';
 import path from 'path';
-import fs from 'fs';
```
Comment (wincent) [inline test/server.js]: Naughty! Sneaking in an unrelated change.

**Comment by wincent** [comment]
Oh, forgot to mention: I applied some small fixes directly on the PR branch rather than incurring another round trip. If you do more work on this, be sure to pull those down.

**Comment by joelgriffith** [comment]
> The fact that the editor views are powered by CodeMirror is really just an implementation detail, and using codeMirror in the prop name leaks that out.

This is a great point, and something that I thought about in some capacity. Here's my thoughts:

- `theme` is a bit broad in my opinion, as new adopters will likely assume that this theme would apply to the entirety of GraphiQL (which isn't the case).
- Minor, but I think that at this point GraphiQL is pretty tied to the CodeMirror implementation (there's even sub-packages that were written to marry GraphiQL to CodeMirror). Not saying that won't change, but would be non-trivial work.

I think maybe `syntaxTheme` would suffice since it succinctly addresses the fact that this only applies to code portions of the UI, and is not a global UI-theme.

Would that be ok @wincent?

**Comment by wincent** [comment]
Or maybe `editorTheme`?

Agreed that for all practical purposes CodeMirror is likely to remain the implementation, but that doesn't mean we shouldn't try to practice good hygiene and abstract over it.

**Comment by joelgriffith** [comment]
:boom: `editorTheme` is great. I'll update my PR tomorrow.

**Comment by wincent** [comment]
Looks good. Thanks for your work on this!

**Comment by joelgriffith** [comment]
Thanks for catching that JSX error....again!
